shared/src/main/scala/squants/market/Money.scala

```diff
@@ -392,10 +392,10 @@ object Money extends Dimension[Money] {
   def apply[A](n: A, currency: Currency)(implicit num: Numeric[A]) = new Money(BigDecimal(num.toDouble(n)))(currency)
   def apply[A](n: A, currency: String)(implicit num: Numeric[A]) = new Money(BigDecimal(num.toDouble(n)))(defaultCurrencyMap(currency))
 
-  def apply(s: String): Try[Money] = {
-    lazy val regex = ("([-+]?[0-9]*\\.?[0-9]+) *(" + defaultCurrencySet.map(_.code).reduceLeft(_ + "|" + _) + ")").r
+  def apply(s: String)(implicit moneyContext: MoneyContext): Try[Money] = {
```

Comment:
I'm not sure about this, passing this as an `implicit` sounds like an overstretch of `implicit`. Wouldn't you rather pass it as a plain param?